src/rgw/rgw_rest_s3.cc

```diff
@@ -1405,6 +1407,9 @@ static int get_success_retcode(int code)
 
 void RGWPutObj_ObjStore_S3::send_response()
 {
+  if(op_ret == 0 && process_callback(s, this, op_ret, etag, boost::none))
```

Comment:
this seems to require being conditional (efficiently for the common case where it's not happening)